nautobot/core/filters.py

```diff
@@ -630,15 +643,36 @@ def _generate_lookup_expression_filters(cls, filter_name, filter_field):
         if field is None:
             return magic_filters
 
+        field_type = None
+        try:
+            field_type = cls._meta.model._meta.get_field(field_name)
+            # TODO: This should not be needed, there seems to be some behavior changes with
+            # RoleFilter filters that haven't been migrated, which is beyond the scope of this change
+            # Working around it for now
+            if isinstance(filter_field, import_string("nautobot.extras.filters.mixins.RoleFilter")):
+                field_type = None
+        except FieldDoesNotExist:
+            # Will get error like:
+            # `FieldDoesNotExist: RelationshipAssociation has no field named 'relationship__key'`
+            pass
+
+        # If the field allows null values, add an `isnull`` check
+        if field.null and isinstance(field_type, NULLABLE_CLASSES):
```

Comment:
Is there a case you encountered where we have a field that is `.null` but *isn't* one of `NULLABLE_CLASSES`? Just wanting to understand what fields we're excluding with the `isinstance` check.

Reply:
I do not know, I asked ChatGPT which ones are, that gave me a list, I added those, you let me know to just use the inherited class, and I figured it was good preventive measure.

Reply:
Just not sure what we're preventing here. If a field has a `null=True` flag on it, it *seems* like we should have a corresponding `isnull` filter on it, regardless of what type of field it is?

Reply:
works for me, I will make the update.

Reply:
It failed on some other similar issue. Can we put this in for now? I think it makes sense to:

* Merge for now
* Develop a system that sanely fails before this becomes an issue, so we can stop such future issues from happening
* Remove the nullable_class check

Reply:
Just change from `if field.null and isinstance(field_type, NULLABLE_CLASSES):` to `if field.null:` and re-run tests to see what I was referring to.

Reply:
The solution with `NULLABLE_CLASSES` works because it skips certain fields. E.g. `dcim.Rack.role` field has `null=True`, but it's skipped when using `NULLABLE_CLASSES`.

Issue is a race condition, as we are trying to fetch data from the database to configure filters before the database is ready.

https://github.com/nautobot/nautobot/blob/develop/nautobot/core/models/fields.py#L190

IMHO it's better to fix this race condition first, and then implement this feature using duck-typing approach. WDYT @glennmatthews?

Reply:
It sounds like my suspicion was correct and we should probably: 

> Develop a system that sanely fails before this becomes an issue, so we can stop such future issues from happening

I realize that I say that without understanding how difficult it may be... but seems like the correct direction.

Reply:
> Issue is a race condition, as we are trying to fetch data from the database to configure filters before the database is ready.
> 
> https://github.com/nautobot/nautobot/blob/develop/nautobot/core/models/fields.py#L190
> 
> IMHO it's better to fix this race condition first, and then implement this feature using duck-typing approach. WDYT @glennmatthews?

Agreed, see also my comment above https://github.com/nautobot/nautobot/pull/4378#discussion_r1334387786

Reply:
Something like this can resolve it for now, tests passes:

```
diff --git a/nautobot/core/filters.py b/nautobot/core/filters.py
index 0851d68fd..ae9b5b017 100644
--- a/nautobot/core/filters.py
+++ b/nautobot/core/filters.py
@@ -5,10 +5,8 @@ import uuid
 
 from django import forms as django_forms
 from django.conf import settings
-from django.core.exceptions import FieldDoesNotExist
 from django.db import models
 from django.forms.utils import ErrorDict, ErrorList
-from django.utils.module_loading import import_string
 import django_filters
 from django_filters.constants import EMPTY_VALUES
 from django_filters.utils import get_model_field, resolve_field
@@ -22,17 +20,6 @@ from nautobot.core.utils import data as data_utils
 
 logger = logging.getLogger(__name__)
 
-NULLABLE_CLASSES = (
-    models.CharField,
-    models.DateField,
-    models.DecimalField,
-    models.ForeignKey,
-    models.FloatField,
-    models.IntegerField,
-    models.TextField,
-    models.UUIDField,
-)
-
 
 def multivalue_field_factory(field_class, widget=django_forms.SelectMultiple):
     """
@@ -643,21 +630,8 @@ class BaseFilterSet(django_filters.FilterSet):
         if field is None:
             return magic_filters
 
-        field_type = None
-        try:
-            field_type = cls._meta.model._meta.get_field(field_name)
-            # TODO: This should not be needed, there seems to be some behavior changes with
-            # RoleFilter filters that haven't been migrated, which is beyond the scope of this change
-            # Working around it for now
-            if isinstance(filter_field, import_string("nautobot.extras.filters.mixins.RoleFilter")):
-                field_type = None
-        except FieldDoesNotExist:
-            # Will get error like:
-            # `FieldDoesNotExist: RelationshipAssociation has no field named 'relationship__key'`
-            pass
-
         # If the field allows null values, add an `isnull`` check
-        if field.null and isinstance(field_type, NULLABLE_CLASSES):
+        if getattr(field, "null", None):
             # Use this method vs extend as the `lookup_map` variable is generally one of
             # the constants which we do not want to update
             lookup_map = dict(lookup_map, **{"isnull": "isnull"})
diff --git a/nautobot/core/models/fields.py b/nautobot/core/models/fields.py
index 49f2bf525..822babb8e 100644
--- a/nautobot/core/models/fields.py
+++ b/nautobot/core/models/fields.py
@@ -187,7 +187,12 @@ class ForeignKeyLimitedByContentTypes(ForeignKeyWithAutoRelatedName):
     """
 
     def get_limit_choices_to(self):
-        return {"content_types": ContentType.objects.get_for_model(self.model)}
+        try:
+            return {"content_types": ContentType.objects.get_for_model(self.model)}
+        except:
+            import logging
+            logging.error(5 * "UNHANDLED", exc_info=True)
+            return {}
 
     def formfield(self, **kwargs):
         """Return a prepped formfield for use in model forms."""
```